crates/cxx-qt-build/src/dependencies.rs

```diff
@@ -0,0 +1,301 @@
+        let path = path
+            .canonicalize()
+            .expect("Failed to canonicalize path to initializer! Does the path exist?");
+        self.initializers.push(path);
+        self
+    }
+
+    /// Export all headers with the given prefix to downstream dependencies
+    ///
+    /// Note: This will overwrite any previously specified header_prefixes, including the default
+    /// header_prefix of this crate.
+    ///
+    /// This function will panic if any of the given prefixes are already exported through the
+    /// [Self::export_include_directory] function.
+    pub fn export_include_prefixes<'a>(
+        mut self,
+        prefixes: impl IntoIterator<Item = &'a str>,
+    ) -> Self {
+        let prefixes = prefixes.into_iter().map(|item| item.to_string()).collect();
+
+        let mut exported_prefixes = self
+            .exported_include_directories
+            .iter()
+            .map(|(_path, prefix)| prefix);
+        for prefix in &prefixes {
+            if let Some(duplicate) =
+                exported_prefixes.find(|exported_prefix| exported_prefix.starts_with(prefix))
+            {
+                panic!("Duplicate export_prefix! Cannot export `{prefix}`, as `{duplicate}` is already exported as an export_include_directory!");
+            }
+        }
+
+        self.exported_include_prefixes = prefixes;
+        self
+    }
+
+    /// Add a directory that will be added as an include directory under the given prefix.
+    ///
+    /// The prefix will automatically be exported (see also: [Self::export_include_prefixes])
+    ///
+    /// This function will panic if the given prefix is already exported.
+    pub fn export_include_directory(mut self, directory: impl AsRef<Path>, prefix: &str) -> Self {
+        let mut exported_prefixes = self.exported_include_prefixes.iter().chain(
+            self.exported_include_directories
+                .iter()
+                .map(|(_path, prefix)| prefix),
+        );
+        if let Some(duplicate) =
+            exported_prefixes.find(|exported_prefix| exported_prefix.starts_with(prefix))
+        {
+            panic!("Duplicate export_prefix! Cannot export `{prefix}`, as `{duplicate}` is already exported!");
+        }
+
+        self.exported_include_directories
+            .push((directory.as_ref().into(), prefix.to_owned()));
+        self
+    }
+
+    /// Reexport the dependency with the given link name.
+    /// This will make the dependency available to downstream dependencies.
+    ///
+    /// Specifically it will reexport all include_prefixes of the given dependency
+    /// as well as any definitions made by that dependency.
+    ///
+    /// Note that the link name may differ from the crate name.
+    /// Check your dependencies manifest file for the correct link name.
+    pub fn reexport_dependency(mut self, link_name: &str) -> Self {
+        self.reexport_links.insert(link_name.to_owned());
+        self
+    }
+}
+
+// cxx-qt-lib example
```

Comment:
either remove these or make them a doc comment / test ?

Also should `python3` be in there?

Reply:
Ah, right, those can go now.

Reply:
Done